packages/server-wallet/src/wallet/index.ts

```diff
@@ -126,18 +126,17 @@ export class Wallet implements WalletInterface {
   }
 
   async pushMessage(message: AddressedMessage): Result {
-    await Store.transaction(async () => {
-      if (message.signedStates) {
-        message.signedStates.forEach(async ss => {
-          await Store.addSignedState(ss);
-        });
-      }
-      if (message.objectives) {
-        message.objectives.forEach(async o => {
-          await Store.addObjective(o);
-        });
-      }
-    });
+    if (message.signedStates) {
+      message.signedStates.forEach(async ss => {
+        await Store.addSignedState(ss);
+      });
+    }
+    if (message.objectives) {
+      message.objectives.forEach(async o => {
+        await Store.addObjective(o);
+      });
+    }
```

Comment:
I think any of these could _return_ an error, not throw an error, so shouldn't this be inside some kind of a loop which checks for errors responses and if it sees one, returns it to the caller?